tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,176 @@
+# Generates RegisterCodegenUnboxedKernels.cpp, UnboxingFunctions.h and UnboxingFunctions.cpp.
+import argparse
+import json
+import os
+from dataclasses import dataclass
+from tools.codegen.api.translate import translate
+from tools.codegen.api.types import CppSignatureGroup
+from tools.codegen.api.unboxing import convert_arguments
+from tools.codegen.context import method_with_native_function
+from tools.codegen.gen import parse_native_yaml
+from tools.codegen.model import NativeFunction, NativeFunctionsGroup, Variant
+from tools.codegen.utils import Target, FileManager, mapMaybe, make_file_manager
+from typing import Union, Sequence
+from typing_extensions import Literal
+
+
+# Generates UnboxingFunctions.h & UnboxingFunctions.cpp.
+@dataclass(frozen=True)
+class ComputeUnboxingFunctions:
+    target: Union[Literal[Target.DECLARATION], Literal[Target.DEFINITION]]
+
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+        sig_group = CppSignatureGroup.from_native_function(
+            f, method=(Variant.method in f.variants), fallback_binding=f.manual_cpp_binding
+        )
+        sig = sig_group.most_faithful_signature()
+
+        if self.target is Target.DECLARATION:
+            # Note [The ATen Codegen Unboxing API]
+            # Similar to the ATen Operators API, ATen Codegen Unboxing API lives in the at::unboxing namespace, and
+            # will be used by codegen unboxing wrappers (CodegenUnboxingWrappers.cpp).
+            # The Wrappers will be registered into torch::jit::OperatorRegistry using RegisterOperators API.
+            #
+            # Important characteristics about the Codegen Unboxing API:
+            # (1) It follows the OperatorRegistry API.
+            #     This is kind of necessary to avoid overhead.
+            #     For example: if it followed the C++ API, then all of the faithful C++ factory functions
+            #     would need to wrap their arguments into TensorOptions only to unwrap them again.
+            # (2) Under the hood it calls C++ API.
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack);
+"""
+        else:
+            # gather all the arguments from native function, including "out"
+            args = list(f.func.arguments.flat_non_out) + (
+                list(f.func.arguments.out) if f.func.arguments.out else []
+            )
+
+            # parse arguments into C++ code
+            binding_list, code_list = convert_arguments(args,
+                                                        has_tensor_options=f.func.arguments.tensor_options is not None)
+
+            # for each C++ argument, generate the conversion code
+            code_connector = "\n\t"
+            arg_connector = ",\n\t\t"
+            # function call and push back to stack
+            prefix = "self_base." if sig.method else "at::"
+            args_str = "" if not binding_list else f"""
```

Comment:
This is to make the generated code a bit more elegant. Instead of having something like
```
auto result_ = self_base.arccos_(
        
);
```
We can have:
```
auto result_ = self_base.arccos_();
```

Reply:
Oh, I'd argue that instead of `arg_connector` having newlines between each arg, you should just put them all on the same line like most other codegen function call-sites. Then you can also remove special-casing logic like that from the codegen